src/main/java/Duke.java

```diff
@@ -1,10 +1,272 @@
+                        System.out.println("Nice! I've marked this task as done:\n  " + tasks.get(index));
+                    } else {
+                        System.out.println("Invalid task index.");
+                    }
+                } else if (command.startsWith("unmark")) {
+                    int index = Integer.parseInt(command.split(" ")[1]) - 1;
+                    if (index >= 0 && index < tasks.size()) {
+                        tasks.get(index).markNotDone();
+                        System.out.println("OK, I've marked this task as not done yet:\n  " + tasks.get(index));
+                    } else {
+                        System.out.println("Invalid task index.");
+                    }
+                }  else if (command.startsWith("delete")) {
+                    int index = Integer.parseInt(command.split(" ")[1]) - 1;
+                    if (index >= 0 && index < tasks.size()) {
+                        Task removedTask = tasks.remove(index);
+                        System.out.println("Noted. I've removed this task:\n  " + removedTask);
+                        System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+                    } else {
+                        System.out.println("Invalid task index.");
+                    }
+                } else {
+                    throw new DukeException("I'm sorry, but I don't know what that means :-(");
+                }
+                saveTasksToFile(tasks);
+            }
+            catch (DukeException e) {
+                System.out.println("☹ OOPS!!! " + e.getMessage());
+            }
+
+            System.out.println("____________________________________________________________");
+        }
+
+
+        scanner.close();
+    }
+    private static String formatDate(LocalDate date) {
+        return date.format(DateTimeFormatter.ofPattern("MMM dd yyyy"));
+    }
+    private static void loadTasksFromFile(ArrayList<Task> tasks) {
+        try {
+            File file = new File(FILE_PATH);
+            if (file.exists()) {
+                Scanner fileScanner = new Scanner(file);
+                while (fileScanner.hasNextLine()) {
+                    String line = fileScanner.nextLine();
+                    Task task = createTaskFromLine(line);
+                    if (task != null) {
+                        tasks.add(task);
+                    }
+                }
+                fileScanner.close();
+            }
+        } catch (FileNotFoundException e) {
+            // Handle file not found exception
+            System.out.println("File not found: " + FILE_PATH);
+        }
+    }
+
+    private static Task createTaskFromLine(String line) {
+        String[] parts = line.split(" \\| ");
+        if (parts.length < 3) {
+            return null;
+        }
+
+        String type = parts[0];
+        String status = parts[1];
+        String description = parts[2];
+
+        Task task = null;
+        switch (type) {
```

Comment:
For switch statements, I think there shouldn't be indentation on the next line. On the mod website:
switch (condition) {
case ABC:
    statements;
    // Fallthrough
case DEF:
    statements;
    break;
case XYZ:
    statements;
    break;
default:
    statements;
    break;
}